### PR TITLE
BrAPI: Fix handling of lastSyncedTime

### PR DESCRIPTION
**Pull request by mobreza**
## Description

This MR fixes the following:

1. After a successful initial "sync" of Study data from a BrAPI 2.1 server the app does not recognize the downloaded Observations as "synced" in the Export view. By setting `lastSyncedTime` after fetching data from the server, this issue is only partially solved. The updated `getStatus()` method takes care of properly assigning `BrapiObservation.Status.SYNCED` vs `EDITED` and the view displays correct numbers.
2. Saving Observations with a `timestamp` or `lastSyncedTime` to the local database failed since the conversion from `OffsetDateTime` to `String` was not used.

image

image

## Type of change

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [X] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Checklist:

- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated relevant documentation

## Changelog entry

```release-note
- Uses current time as lastSyncedTime when downloading Observations from the server
- Updates lastSyncedTime after creating/updating Observations
- Fixes issues with saving "observation_time_stamp" and "last_synced_time" in DAO
- Updates BrAPIObservation#getStatus method to correctly identify synced and updated records
- Improved support for BrAPI servers not using https
```

### CODE REVIEW

**Comment by chaneylc** [comment]
Hello @mobreza thanks for the contribution :)

We have just released a new version, with it came some changes that may have affected this PR. Could you please test this branch again to make sure your functionality is working?

File: app/src/main/java/com/fieldbook/tracker/charts/HistogramChartHelper.kt [hunk 1]
```diff
@@ -190,7 +190,7 @@ object HistogramChartHelper {
         Log.d(TAG, "Max bin count: $maxBinCount, Ideal bin count: $idealBinCount, Final bin count: ${minOf(maxBinCount, idealBinCount)}")
 
         // Use the lesser of maxBinCount and idealBinCount
-        return minOf(maxBinCount, idealBinCount)
+        return maxOf(1, minOf(maxBinCount, idealBinCount)) // at least one bin
```
Comment (mobreza) [inline app/src/main/java/com/fieldbook/tracker/charts/HistogramChartHelper.kt:193]: @chaneylc I had a case where the calculated `idealBinCount` was 0 due to `(range/binWidthFD).toInt()` causing the app to crash with DIV/0. I'm not sure if this is the best place to resolve the issue.

**Comment by chaneylc** [comment]
@mobreza Thanks for looking at this so quickly, is it ok to merge?

**Comment by mobreza** [comment]
When I pick the "Field" to download from the server it gives me the following snack:

image

and none of the details are downloaded. But if I then open the "Field" and run a sync it's happy as a clam. No clue where that comes from.

**Review by chaneylc — APPROVED** [review]

**Comment by mobreza** [comment]
This might help:

https://github.com/user-attachments/assets/2bb42b24-094d-4afb-9ce7-82c97feb621d

**Comment by chaneylc** [comment]
The false error message is fixed now, thanks again for your help on this :)

**Pull request merged**